include/revng/ADT/STLExtras.h

```diff
@@ -53,3 +53,54 @@ concept IsNotTupleEnd
 static_assert(HasTupleSize<std::tuple<>>);
 static_assert(!HasTupleSize<std::vector<int>>);
 static_assert(!HasTupleSize<int>);
+
+//===----------------------------------------------------------------------===//
+//     Extra additions to <iterator>
+//===----------------------------------------------------------------------===//
+
+namespace revng {
+namespace detail {
+template<typename ItTy,
+         typename FuncTy,
+         typename FuncReturnTy = decltype(
+           std::declval<FuncTy>()(*std::declval<ItTy>()))>
+class proxy_mapped_iterator_impl : public llvm::mapped_iterator<ItTy, FuncTy> {
+  struct IteratorProxy {
+    FuncReturnTy Temporary;
```

Comment:
I really don't want `Temporary` to be accessible ever, so it should be `private`.